File: Data Structures & Algorithms/car-fleet/submission-4.java
class Solution {
    public int carFleet(int target, int[] position, int[] speed) {
        int res = 0;

        int[][] pair = new int[position.length][2];
        for (int i = 0; i < position.length; i++) {
            pair[i][0] = position[i];
            pair[i][1] = speed[i];
        }
        Arrays.sort(pair, (a, b) -> Integer.compare(b[0], a[0]));

        double prev = 0;
        for(int i = 0; i < pair.length; i++){
            int pos = pair[i][0];
            int sp = pair[i][1];

            double temp = (double) (target - pos) / sp;

            System.out.println("Pos: " + pos + " Sp: " + sp + " Temp: " + temp + " Prev: " + prev);

            if(temp > prev){
                res += 1;
                prev = temp;
            }
        }

        return res;
    }
}

File: Data Structures & Algorithms/is-palindrome/submission-4.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        if len(s) < 2: return True
        l, r = 0, len(s) - 1
        res = False

        while l < r:
            while not self.isAlphaNumeric(s[l]) and l < r:
                l += 1

            while not self.isAlphaNumeric(s[r]) and l < r:
                r -= 1

            if s[l].lower() != s[r].lower(): return False
            res = True

            l += 1
            r -= 1

        return res

    def isAlphaNumeric(self, c):
        return (
            ('a' <= c <= 'z') or 
            ('A' <= c <= 'Z') or
            ('0' <= c <= '9')
               )
        